atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplier.java

```diff
@@ -73,6 +79,35 @@ public KeyValueService getKeyValueService() {
         return keyValueService.get();
     }
 
+    private KeyValueService pollOrCreateRawKeyValueService(AtlasDbFactory atlasFactory) {
+        return atlasDbConfig.pollForDataBaseOnStartup() ? pollForKeyValueService(atlasFactory)
+                : atlasFactory.createRawKeyValueService(keyValueServiceConfig, leaderConfig);
+    }
+
+    private KeyValueService pollForKeyValueService(AtlasDbFactory atlasFactory) {
+        int failureCount = 0;
```

Comment:
This impl is pretty clean/concise so don't feel too strongly on Awaitility - mostly would just like to see the the backoff logic separated out into its own class.